src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js

```diff
@@ -1,56 +1,100 @@
+import { convertValue as convertServerToClient } from '../../converters/serverToClient';
 import type { Props } from './widgetProfile.types';
-import { useProfileInfo } from './hooks';
+
 
 const styles = {
-    flatListWrapper: {
-        padding: '0 16px',
-    },
-    itemRow: {
-        borderBottom: `1px solid${colors.grey400}`,
-        display: 'flex',
-        padding: '16px 0',
-        '&.isLastItem': {
-            borderBottomWidth: 0,
-        },
-    },
-    itemKey: {
-        width: 150,
-        color: colors.grey600,
+    profileWidgetWrapper: {
+        paddingBottom: '12px',
     },
+
 };
-const ProfileWidgetPlain = ({ classes }: Props) => {
-    const { attributes } = useProfileInfo();
+const ProfileWidgetPlain = ({ classes, teiId, programId }: Props) => {
     const [open, setOpenStatus] = useState(true);
-    const lastItem = attributes[attributes.length - 1];
+    const getProgram = useMemo(() => ({
+        indicators: {
+            resource: 'programs',
+            id: programId,
+            params: {
+                fields:
+                ['programTrackedEntityAttributes[id,displayInList,trackedEntityAttribute[id,displayName,valueType]]'],
+            },
+        },
+    }), [programId]);
+
+    const getAttributes = useMemo(() => ({
+        indicators: {
+            resource: 'trackedEntityInstances',
+            id: teiId,
+            params: {
+                program: programId,
+            },
+        },
+    }), [teiId, programId]);
+
+
+    const { loading: programLoading, data: programQuery, error: programError } = useDataQuery(getProgram);
+    const { loading: attributesLoading, data: attributesQuery, error: attributesError } = useDataQuery(getAttributes);
+    const loading = programLoading || attributesLoading;
+    const error = programError || attributesError;
+
+    if (loading) {
+        return <></>;
+    }
+
+    if (error) {
+        throw error;
+    }
+
+    const mergeAttributes = () => {
+        const { programTrackedEntityAttributes } = programQuery.indicators;
+        const { attributes } = attributesQuery.indicators;
+        const displayEntities = programTrackedEntityAttributes
+            .filter(item => item.displayInList)
+            .reduce((acc, curr) => { acc = [...acc, curr.trackedEntityAttribute]; return acc; }, []);
+
+        const formattedAttributes = [];
+        displayEntities.forEach((entity) => {
+            const displayAttribute = attributes.find(att => att.attribute === entity.id);
+            if (displayAttribute) {
+                formattedAttributes.push(
+                    { ...entity,
+                        value: convertServerToClient(displayAttribute.value, displayAttribute.valueType),
+                    });
+            }
+        });
+        return formattedAttributes
+            .map(attribute => (
+                { id: attribute.id,
+                    key: attribute.displayName,
+                    children: <>
```

Comment:
In case we want to display the image (for the photograph) I think it's better to create the element here to keep the flatlist generic. 
Another approach is to pass the value to the flat list and create the element there. What do you prefer?

Reply:
My preferred approach to this would be to allow for what flow defines as Node (could be number, text, react element etc). Anything defined as Node can be rendered by React (hence, node is the return type for the functional components and the return type of the render function of the class component). https://flow.org/en/docs/react/types/#toc-react-node

For images the convertMethod will return a react element itself.